packages/shared-core/src/utils.ts

```diff
@@ -1,18 +1,27 @@
 import {
   LegacyFilter,
-  SearchFilterGroup,
-  FilterGroupLogicalOperator,
+  UISearchFilter,
+  UILogicalOperator,
   SearchFilters,
   BasicOperator,
   ArrayOperator,
   isLogicalSearchOperator,
+  SearchFilter,
 } from "@budibase/types"
 import * as Constants from "./constants"
-import { removeKeyNumbering } from "./filters"
-
-// an array of keys from filter type to properties that are in the type
-// this can then be converted using .fromEntries to an object
-type AllowedFilters = [keyof LegacyFilter, LegacyFilter[keyof LegacyFilter]][]
+import { removeKeyNumbering, splitFiltersArray } from "./filters"
+import _ from "lodash"
+
+const FILTER_ALLOWED_KEYS = [
```

Comment:
I guess this is the subset of LegacyFilter, right? Could we type the `const _FILTER_ALLOWED_KEYS: (keyof LegacyFilter)[]`, to help removing magic strings?

Reply:
Great call, will do!

Reply:
This is actually quite interesting. It calls out `valueType` and `formulaType` as not being keys of `SearchFilter` (can't do `LegacyFilter` because that includes the allOr and onEmptyFilter objects, and makes the type functionally a `never[]`).

If I add them to the type, they won't be used anywhere in TS code but they will be in JS code.